include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -220,13 +220,72 @@ namespace internal
 
     // helper function
     const auto sort_constraints = [&]() {
+      if (locally_relevant_constraints.empty())
+        return;
+
       std::sort(locally_relevant_constraints.begin(),
                 locally_relevant_constraints.end());
 
-      locally_relevant_constraints.erase(
-        std::unique(locally_relevant_constraints.begin(),
-                    locally_relevant_constraints.end()),
-        locally_relevant_constraints.end());
+      auto equal_with_tol = [](const number d, const number e) {
+        if (std::abs(std::real(d - e)) <
+            100000. *
```

Comment:
The tolerance which is used for the checks if an entry or the inhomogeneity is equal is up for discussion.